R/operate.R

```diff
@@ -0,0 +1,110 @@
+#' Arithmetic operations for messydates
+#'
+#' These operations allow users to add or subtract dates messydate objects.
+#' Messydate objects include incomplete or uncertain dates,
+#' ranges of dates, negative dates, and date sets.
+#' @param e1 A messydate object
+#' @param e2 A numerical object.
+#' @return A messydates vector
+#' @examples
+#' d <- as_messydate(c("2008-03-25", "-2012-02-27", "2001-01?", "~2001",
+#' "2001-01-01..2001-02-02", "{2001-01-01,2001-02-02}",
+#' "2008-XX-31", "..2002-02-03", "2001-01-03..", "28 BC"))
+#' tibble::tibble(date = d, add = d + 1, subtract = d - 1)
+#' tibble::tibble(date = d, add = d + "1 year", subtract = d - "1 year")
+#' @name operate
+NULL
+
+#' @rdname operate
+#' @export
+`+.messydt` <- function(e1, e2) {
+  e2 <- parse_date_strings(e2)
+  add(e1, e2)
+}
+
+#' @rdname operate
+#' @export
+`-.messydt` <- function(e1, e2) {
+  e2 <- parse_date_strings(e2)
+  subtract(e1, e2)
+}
+
+#' Helper function to add to messydates
+#'
+#' @param x A mssydate or list of messydates
+#' @param n What do you want to be added?
+#' Please specify in numeric format (i.e. one day is 1).
+#' @importFrom stringr str_detect str_replace str_split
+#' @importFrom lubridate as_date
+#' @details `add()` adds date units to dates, negative dates,
+#' ranges of dates, and sets of dates.
+#' For ranges of dates, it adds the unit on each side.
+#' @return A messydates vector
```

Comment:
Do we need documentation for these if they are just helper functions/not user facing?